lektor_tags.py

```diff
@@ -148,3 +166,103 @@ def get_all_tags(self, parent):
 
     def ignore_missing(self):
         return bool_from_string(self.get_config().get("ignore_missing"), False)
+
+    def _get_tagcount(self):
+        """Return a dictionary of tag counts.
+
+        Compute it only once, and return the "cached" version then.
+        """
+        if self._tagcount is not None:
+            return self._tagcount
+
+        # Count tags, to be aggregated as "tag weights". Note that tags that
+        # only appear in non-discoverable pages are ignored.
+        self._tagcount = collections.defaultdict(int)
+        for page in get_ctx().pad.query(self.get_parent_path()):
+            for tag in page[self.get_tag_field_name()]:
+                self._tagcount[tag] += 1
+        return self._tagcount
+
+    def _get_minmax_count(self):
+        """Return a tuple of minimum and maximum tag counts."""
+        tagcount = self._get_tagcount()
+        if tagcount:
+            return min(tagcount.values()), max(tagcount.values())
+        else:
+            # There isn't a single tag
+            return 0, 0
+
+    def _weight_count(self):
+        """Map each tag to the number of pages tagged with it."""
+        return self._get_tagcount()
+
+    def _weight_linear(self, a, b):
+        """Map each tag with a number between `a` and `b`.
+
+        The less used tag is mapped `a`, the most used tag is mapped `b`.
```

Comment:
less -> least?

Reply:
Done in 1e9d758e3284df9b3cc4504b21a7a0db8c4e7d08